### PR TITLE
First status

### PR DESCRIPTION
**Pull request by TomChapmanGov**
#1107

### CODE REVIEW

**Review by dinesh-aot — APPROVED** [review]

**Comment by dinesh-aot** [comment]
Please resolve the conflicts

File: epictrack-web/src/components/shared/index.tsx [hunk 1]
```diff
@@ -425,3 +453,28 @@ export const ETPreviewBox = ({ children, sx, ...rest }: HeaderProps) => {
     </Box>
   );
 };
+
+interface DashedBorder {
+  children?: React.ReactNode | string;
+  [prop: string]: unknown;
+  sx?: SxProps;
+}
+export const DashedBorder = ({ children, sx, ...rest }: DashedBorder) => {
```
Comment (jadmnsaad) [inline epictrack-web/src/components/shared/index.tsx]: I've actually added the preview text and previewbox for this? Could you check them?

File: epictrack-web/src/components/work/workStaff/WorkStaffList.tsx [hunk 2]
```diff
@@ -110,6 +110,8 @@ const WorkStaffList = () => {
     [wsData]
   );
 
+  console.log(teamFilter);
```
Comment (jadmnsaad) [inline epictrack-web/src/components/work/workStaff/WorkStaffList.tsx]: Debug

File: epictrack-web/src/components/workPlan/status/StatusContext.tsx [hunk 3]
```diff
@@ -1,18 +1,146 @@
...
+
+  // React.useEffect(() => {
+  //   setStatus(testStatus);
+  // }, []);
+
+  const onDialogClose = () => {
+    setShowStatusForm(false);
+  };
+
+  const onSave = React.useCallback(
```
Comment (jadmnsaad) [inline epictrack-web/src/components/workPlan/status/StatusContext.tsx]: I don't think we need a useCallback here. onSave could be just a simple arrow function

File: epictrack-web/src/components/workPlan/status/StatusForm.tsx [hunk 4]
```diff
@@ -26,8 +39,15 @@ const StatusForm = () => {
...
   } = methods;
 
-  const onSubmitHandler = async (data: Status) => {
-    return null;
+  const handleDescriptionChange = (event: any) => {
+    setNotes(event.target.value);
+  };
+
+  const onSubmitHandler = async (data: any) => {
+    console.log(data);
```
Comment (jadmnsaad) [inline epictrack-web/src/components/workPlan/status/StatusForm.tsx]: debug

File: epictrack-web/src/components/workPlan/status/StatusForm.tsx [hunk 5]
```diff
@@ -41,7 +61,63 @@ const StatusForm = () => {
...
+      >
+        <Grid item xs={4}>
+          <ETFormLabel required>Date</ETFormLabel>
+          <Controller
+            name="start_date"
+            control={control}
+            defaultValue={Moment(status?.start_date).format()}
+            render={({ field: { onChange, value }, fieldState: { error } }) => (
+              <LocalizationProvider dateAdapter={AdapterDayjs}>
+                <DatePicker
```
Comment (jadmnsaad) [inline epictrack-web/src/components/workPlan/status/StatusForm.tsx:72]: do we need a date picker for this or a textfield with type='date'?

File: epictrack-web/src/components/workPlan/status/StatusView/RecentStatus.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+const RecentStatus = () => {
+  const { statuses } = React.useContext(WorkplanContext);
+  const { setShowStatusForm, setStatus, setShowApproveStatusDialog } =
+    React.useContext(StatusContext);
+  const { position } = useAppSelector((state) => state.user.userDetail);
+
+  return (
+    <Box
+      sx={{
```
Comment (jadmnsaad) [inline epictrack-web/src/components/workPlan/status/StatusView/RecentStatus.tsx:25]: should we use the graybox component here?

File: epictrack-web/src/components/workPlan/status/StatusView/RecentStatus.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+            sx={{
+              color: Palette.error.dark,
+              backgroundColor: Palette.error.bg.light,
+              padding: "4px 8px",
+              borderRadius: "4px",
+            }}
+          >
+            Need Approval
+          </ETCaption1>
+        ) : (
```
Comment (jadmnsaad) [inline epictrack-web/src/components/workPlan/status/StatusView/RecentStatus.tsx]: Check out the react-if library, we have it installed and we used it a few times. If you liked it you could use it here, could be more readable

File: epictrack-web/src/components/workPlan/status/StatusForm.tsx (hunk 5 above)
Comment (TomChapmanGov) [reply epictrack-web/src/components/workPlan/status/StatusForm.tsx:72]: i used datepicker so it looks same as the rest of our date inputs

File: epictrack-web/src/components/workPlan/status/StatusView/RecentStatus.tsx (hunk 7 above)
Comment (TomChapmanGov) [reply epictrack-web/src/components/workPlan/status/StatusView/RecentStatus.tsx]: willcheck this out

**Pull request merged**